### ISSUE TITLE
Are seed vs random_state parameters identical?

### ISSUE THREAD

**Issue by iamharshit** [dmlc/xgboost#11330]
I am using XGBRegressor and I see we have 2 possible parameters to control randomness of XGBoost algorithm.

1. `random_state` - https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.XGBRegressor
2. `seed` - https://xgboost.readthedocs.io/en/stable/parameter.html#learning-task-parameters

From documentation there seems no difference between two. 
May I confirm, if these params are indeed identical in functionality?

**Comment by trivialfis** [comment]
For the sklearn interface `XGBRegressor` can handle inputs like `np.random.Generator` https://github.com/dmlc/xgboost/blob/58e908c6649a166c3b47ea5e06e2a8539718a2b0/python-package/xgboost/sklearn.py#L986

Other than this, they are the same.
